### ISSUE TITLE
Typo? "7B require ~26 GB of GPU memory (A100 GPU)."

### ISSUE THREAD

**Issue by diegomontoya** [Lightning-AI/lit-llama#67]
Is the following a typo or the lit-llama implementation requires vastly more vram than original implementation? 7B fits natively on a single 3090 24G gpu in original llama implementation. 

```
This will run the 7B model and require ~26 GB of GPU memory (A100 GPU).
```

**Comment by lantiga** [comment]
Without any quantization 7B float32 parameters means
```
7e9 (7B) * 4 (4 bytes per float32 parameter) / 1024**3 (bytes to GB) = 26.07 GB
```
so it checks out.

You can quantize the model as shown in the finetuning example and you can make it fit in a lot less memory of course, but vanilla will take you 26GB just out of the parameter count (the original repo is not different in that regard).

**Comment by diegomontoya** [comment]
The hugging faces (native) conversion of the model is fp16 and use about 14GB of vram. There was no quantizing performed for the huggingfaces model.  

https://huggingface.co/decapoda-research/llama-7b-hf/blob/main/config.json

Or am I mistaken and they quantized it from fp32 to fp16?

**Comment by lantiga** [comment]
oh for sure, you can run it with `fp16` or `bf16` mixed precision super easily already with `Fabric`. In fact, in `train.py` you can see at line 45

```
 fabric = L.Fabric(accelerator="cuda", devices=4, precision="bf16-mixed", strategy=strategy)
```

which runs it with bf16 mixed precision already and decreases the memory requirements.

In the README we are referring to the vanilla model in full precision which we use to compare against the original llama repo.

We should definitely make the README clearer there, thanks for pointing that out.

/cc @awaelchli

**Comment by carmocca** [comment]
Another missing piece here is loading directly with a lower precision. Opened #71

**Comment by lantiga** [comment]
@diegomontoya for clarity: for inference, you if you just half the precision in generate.py and then move to GPU
```
model.to(device="cuda", dtype=torch.float16)
```
then you'll get 14GB memory usage even with the original checkpoints

**Comment by lantiga** [comment]
Putting this on your radar @diegomontoya  https://github.com/Lightning-AI/lit-llama/pull/91

**Comment by lantiga** [comment]
Solved with #100
